core/src/autogluon/core/trainer/abstract_trainer.py

```diff
@@ -3046,3 +3061,62 @@ def calibrate_model(self, model_name: str = None, lr: float = 0.01, max_iter: in
                 logger.log(15, f'Temperature term found is: {temp_scalar}')
                 model.params_aux["temperature_scalar"] = temp_scalar
                 model.save()
+
+    def calibrate_decision_threshold(self,
+                                     X: pd.DataFrame = None,
+                                     y: np.array = None,
+                                     metric: Union[str, Scorer] = None,
+                                     model: str = 'best',
+                                     weights=None,
+                                     decision_thresholds: Union[int, List[float]] = 50,
+                                     verbose: bool = True) -> float:
+        # TODO: Docstring
+        assert self.problem_type == BINARY, f'calibrate_decision_threshold is only available for `problem_type="{BINARY}"`'
+
+        if metric is None:
+            metric = self.eval_metric
+        elif isinstance(metric, str):
+            metric = get_metric(metric, self.problem_type, 'eval_metric')
+
+        if model == 'best':
+            model = self.get_model_best()
+
+        if y is None:
+            # If model is refit_full, use its parent to avoid over-fitting
+            model_parent = self.get_refit_full_parent(model=model)
+            if not self.model_exists(model_parent):
+                raise AssertionError(f'Unable to calibrate the decision threshold on the internal data because the '
+                                     f'model "{model}" is a refit_full model trained on all of the internal data, '
+                                     f'whose parent model "{model_parent}" does not exist or was deleted.\n'
+                                     f'It may have been deleted due to `predictor.fit(..., keep_only_best=True)`. '
+                                     f'Ensure `keep_only_best=False` to be able to calibrate refit_full models.')
+            model = model_parent
+            if self.has_val:
+                # Use validation data
+                X = self.load_X_val()
+                if self.weight_evaluation:
+                    X, weights = extract_column(X=X, col_name=self.sample_weight)
+                y = self.load_y_val()
+                y_pred_proba = self.predict_proba(X=X, model=model)
+            else:
+                # Use out-of-fold data
+                if self.weight_evaluation:
+                    X = self.load_X()
```

Comment:
Yes it is wiped when optimize for deployment, so it would crash. With that said, we don't have helpful logging in this situation for a lot of places, so doing it properly would be a non-trivial effort. I added a todo for this. I think most users won't run into this problem though, and specifying `calibrate_decision_threshold=True` in the fit call will ensure that the calibration occurs before the data is deleted.